index.d.ts

```diff
@@ -12,23 +13,13 @@
  * Other than `type`, the structure of an action object is really up to you.
  * If you're interested, check out Flux Standard Action for recommendations on
  * how actions should be constructed.
+ *
+ * @template T the type of the action's `type` tag.
  */
-export interface Action {
-  type: any;
-}
-
-/**
- * An Action type which accepts any other properties.
- * This is mainly for the use of the `Reducer` type.
- * This is not part of `Action` itself to prevent users who are extending `Action.
- * @private
- */
-export interface AnyAction extends Action {
```

Comment:
Are you sure `AnyAction` can be safely removed? Its absence broke stuff on TS 2.4

See https://github.com/reactjs/redux/pull/2467